JD Client README file retouched (#800)
stratum-mining/stratum: roles/jd-client/README.md

# JD Client

* connect to the jd-server
* connect to the template-provider
The JD Client receives custom block templates from a Template Provider and declares use of the template with the pool using the Job Declaration Protocol. Further distributes the jobs to Mining Proxy (or Proxies) using the Job Distribution Protocol. ```
* transparently relay the `OpenExtendedChannel` to upstream 

## Setup

### Configuration File

The configuration file contains the following information:

1. The downstream socket information, which includes the listening IP address (`downstream_address`) and port (`downstream_port`).
2. The maximum and minimum SRI versions (`max_supported_version` and `min_supported_version`) with size as (`min_extranonce2_size`)

> 4. The authentication keys for open encrypted connection for the downstream (`authority_public_key`, `authority_secret_key` and `cert_validity_sec`)

Reviewer: ```suggestion
3. The authentication keys for the downstream connection (`authority_public_key`, `authority_secret_key`)
```

> 5. The retry that tells JDC the number of times to retry itself after a failure.

Reviewer: ```suggestion
5. A `retry` parameter which tells JDC the number of times to reinitialize itself after a failure.
```

> 5. A `retry` parameter which tells JDC the number of times to reinitialize itself after a failure.

Reviewer: ```suggestion
4. A `retry` parameter which tells JDC the number of times to reinitialize itself after a failure.
```

6. The Template Provider address (`tp_address`).
7. Optionally, you may want to verify that your TP connection is authentic. You may get `tp_authority_public_key` from the logs of your TP, for example:

> 

Reviewer: ```suggestion
```
# 2024-02-13T14:59:24Z Template Provider authority key: EguTM8URcZDQVeEBsM4B5vg9weqEUnufA8pm85fG4bZd
```
```

> 6. The Job Declarator information includes the Template Provider connection address to which to connect (`tp_address`).
> 7. You may want to verify that your TP connection is authentic. You may get `tp_authority_public_key` from the logs of your.

Reviewer: ```suggestion
6. The Template Provider address (`tp_address`).
7. Optionally, you may want to verify that your TP connection is authentic. You may get `tp_authority_public_key` from the logs of your TP, for example:
```
# 2024-02-13T14:59:24Z Template Provider authority key: EguTM8URcZDQVeEBsM4B5vg9weqEUnufA8pm85fG4bZd
```
```

# JD Client

* connect to the jd-server
* connect to the template-provider
The JD Client receives custom block templates from a Template Provider and declares use of the template with the pool using the Job Declaration Protocol. Further distributes the jobs to Mining Proxy (or Proxies) using the Job Distribution Protocol. ```
* transparently relay the `OpenExtendedChannel` to upstream 

## Setup

### Configuration File

The configuration file contains the following information:

> 1. The downstream connection information includes  connection address (`downstream_address`) and port (`downstream_port`).

Reviewer: ```suggestion
1. The downstream socket information, which includes the listening IP address (`downstream_address`) and port (`downstream_port`).
```

# JD Client

* connect to the jd-server
* connect to the template-provider

> * listen for an `OpenExtendedChannel` from downstream

Reviewer: ```suggestion
The JD Client receives custom block templates from a Template Provider and declares use of the template with the pool using the Job Declaration Protocol. Further distributes the jobs to Mining Proxy (or Proxies) using the Job Distribution Protocol. ```

* transparently relay the `OpenExtendedChannel` to upstream 

> After the setup phase, it will start to negotiate jobs upstream and send the negotiated job
> downstream, so that everything downstream does not need to know that the job is negotiated.

Reviewer: ```suggestion
```
